### DISCUSSION TITLE
fit-time & conditions

### DISCUSSION THREAD

**Discussion by TinusUU** [mwshinn/PyDDM#77] [Ask for help] [answered]
Hi, 

If I have a coherence condition ['coh'] like in the Roitman sample, would ['coh'] be treated as one 'condition' when it is scaled with drift-rate (like in the Roitman example: self.driftcoh * conditions['coh'] ) or would it still be seen as three separate conditions, like in the case when one will fit a separate drift-rate (coh1, coh2, coh3) for each {coh: [0.128, 0.256, 0.512]}?

In the first (scaling) case, where coherence is used as a 'regressor' on drift-rate, would adding more coherences make pyDDM slower? (adding more conditions will -of course- affect speed: https://pyddm.readthedocs.io/en/stable/faqs.html)

Thanks,
Martijn

**Comment by mwshinn** [comment, marked as answer]
The "condition" is the name PyDDM uses for a variable which can change from trial to trial.  So there is only one condition ("coh") but it can take on many values.  So everything depends on how you define the model.  In the case in the quickstart guide, the value of the "coh" condition is multiplied by the "driftcoh" parameter (as you mentioned in your question).  So there is one parameter that can be fit ("driftcoh") and one which can be different on each trial.

As to whether it will make it slower, it depends.  In general, the answer is "yes, a little bit slower per condition".  If you add one or two more coherences, there will be little to no difference in practice.  If you add lots of coherences (10-100) then yes, it will make it slower.  If you have parallelization turned on (set_N_cpus), then it will only become slower if the number of coherences exceeds the number of CPUs.  As a general rule of thumb, PyDDM has to solve the equations once per condition found in the dataset.
